### PR TITLE
frogger-game

### PR DESCRIPTION
**Pull request by KonstantinOkhorzin**
### Object Oriented JS

[Demo](https://konstantinokhorzin.github.io/frontend-nanodegree-arcade-game/) |
[Code base](https://github.com/KonstantinOkhorzin/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

File: submissions/KostiantynOkhorzin/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+        if (this.x > 510) {
+            this.x = -50;
+            this.speed = Math.round(Math.random() * 100 + 100); 
+        };
+
+        if (player.x < this.x + 80 && player.x + 80 > this.x &&
+            player.y < this.y + 60 && 60 + player.y > this.y) {
+            player.x = 200;
+            player.y = 400;
```
Comment (OleksiyRudenko) [inline submissions/KostiantynOkhorzin/frogger-game/app.js:20]: As a peer developer that needs to contribute to this project I want to understand semantics behind all these magic numbers here and below in the code.

Also apparently `update` method doesn't just updates position but also has some more of business logic. We need to separate concerns. Please move checks to a different method.

File: submissions/KostiantynOkhorzin/frogger-game/app.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            this.y += 80;
+        };
+    }
+}
+
+const player = new Player(200, 400);
+
+const enemy1 = new Enemy(-100, 60, 100);
+const enemy2 = new Enemy(-50, 140, 150);
+const enemy3 = new Enemy(-150, 230, 200);
```
Comment (OleksiyRudenko) [inline submissions/KostiantynOkhorzin/frogger-game/app.js:70]: As a peer developer on the project I am tasked to add 100 more rows to the field and 100 more enemies. There should be a way to do it programmatically rather than by adding another 100 lines of code.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@KonstantinOkhorzin good start!
A few improvements are required tho.

**Comment by KonstantinOkhorzin** [comment]
I fixed all. Please, check the changes.

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
This comment remains unadressed: https://github.com/kottans/frontend-2022-homeworks/pull/158#discussion_r946198369

What are all these magic numbers?

**Comment by KonstantinOkhorzin** [comment]
I redid without magic numbers. Please, check the changes.

File: submissions/KostiantynOkhorzin/frogger-game/app.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        return Math.round(Math.random() * (config.enemy.maxSpeed - config.enemy.minSpeed) + config.enemy.minSpeed);
+    }
+
+    checkCollision() {
+        if(player.x < this.x + config.canvas.blockWidth &&
+            player.x + config.canvas.blockWidth > this.x &&
+            player.y < this.y + config.canvas.blockHeight && 
+            config.canvas.blockHeight + player.y > this.y) {
+            player.x = config.player.initialPositionOnX;
+            player.y = config.player.initialPositionOnY;
```
Comment (OleksiyRudenko) [inline submissions/KostiantynOkhorzin/frogger-game/app.js:46]: Classes should not refer to any global variables, like global variable `player`, which is an instance of `Player` class (referring to global constants and globals provided by the gaming platform like Resources is OK); Hint: pass `Player` instance as an argument to every enemy

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@KonstantinOkhorzin well done! Now the code reads almost in plain English.

Yet one thing to make the code OOP/SOLID compliant.

**Comment by KonstantinOkhorzin** [comment]
I fixed this bug. Please, check the changes.

**Review by OleksiyRudenko — APPROVED** [review]
@KonstantinOkhorzin great job!
